src/spell/modules/spell/spell/src/log.rs

```diff
@@ -0,0 +1,189 @@
+use eyre::WrapErr;
+use marine_rs_sdk::marine;
+use marine_sqlite_connector::State;
+
+use fluence_spell_dtos::value::UnitValue;
+
+use crate::auth::{is_by_creator, is_by_spell};
+use crate::schema::db;
+
+
+#[marine]
+pub struct AllLogsResult {
+    pub logs: Vec<String>,
+    pub success: bool,
+    pub error: String,
+}
+
+#[marine]
+pub fn store_log(log: String) -> UnitValue {
+    let call_parameters = marine_rs_sdk::get_call_parameters();
+
+    // We want to prevent anyone except this spell to store logs to its kv
+    if !is_by_creator() || !is_by_spell(&call_parameters) {
+        return UnitValue::error("store_log can be called only by the associated spell script");
+    }
+
+    let result: eyre::Result<()> = try {
+        let conn = db();
+        let mut statement = conn.prepare(
+            r#"
+        INSERT INTO logs
+            (log)
+        VALUES
+            (?)
+        "#,
+        )?;
+        statement.bind(1, log.as_str())?;
+
+
+        statement.next()?;
+    };
+
+    match result {
+        Ok(_) => UnitValue::ok(),
+        Err(e) => UnitValue::error(format!("Error storing log: {}", e)),
+    }
+}
+
+#[marine]
+pub fn get_logs() -> AllLogsResult {
+    let result: eyre::Result<Vec<String>> = try {
+        let conn = db();
+        let mut statement = conn.prepare(
+            r#"
+            SELECT
+                log
+            FROM
+                logs
+        "#,
+        )?;
+        std::iter::from_fn(move || {
+            let r: eyre::Result<Option<String>> = try {
+                if let State::Row = statement.next()? {
+                    Some(statement.read::<String>(0)?)
+                } else {
+                    None
+                }
+            };
+            r.context("error fetching log row from sqlite")
+                .transpose()
+        })
```

Comment:
I wonder if it's possible to DRY this. I think we use that pattern in a lot of places.